src/libs/ReportUtils.ts

```diff
@@ -1868,8 +1871,18 @@ function getDisplayNameForParticipant(accountID?: number, shouldUseShortForm = f
     }
 
     const personalDetails = getPersonalDetailsForAccountID(accountID);
-    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-    const formattedLogin = LocalePhoneNumber.formatPhoneNumber(personalDetails.login || '');
+    if (!personalDetails) {
+        return '';
+    }
+
+    // Check if the phone number is already cached
+    let formattedLogin = phoneNumberCache[personalDetails.login ?? ''];
+    if (!formattedLogin) {
+        formattedLogin = LocalePhoneNumber.formatPhoneNumber(personalDetails.login ?? '');
+        // Store the formatted phone number in the cache
+        phoneNumberCache[personalDetails.login ?? ''] = formattedLogin;
```

Comment:
NIT: The `personalDetails.login ?? '' ` gets used three times. What do you think about extracting it into variable?

Reply:
done